openmdao/utils/coloring.py

```diff
@@ -395,27 +391,75 @@ def _check_config_partial(self, system):
-                               "configuration of the current driver. Make sure you don't have "
-                               "different problems that have the same coloring directory.  Set "
-                               "the coloring directory by setting the value of "
-                               "`problem.options['coloring_dir']`." % system.pathname)
+            ordered_wrt_info = list(system._jacobian_wrt_iter(info['wrt_matches']))
+
+        of_names = [t[0] for t in ordered_of_info]
+        wrt_names = [t[0] for t in ordered_wrt_info]
+
+        of_sizes = [t[2] - t[1] for t in ordered_of_info]
+        wrt_sizes = [t[2] - t[1] for t in ordered_wrt_info]
+
+        self._config_check_msgs(of_names, of_sizes, wrt_names, wrt_sizes, system)
+
+    def _config_check_msgs(self, of_names, of_sizes, wrt_names, wrt_sizes, obj):
+        msg_suffix = ("Make sure you don't have different problems that have the same coloring "
+                      "directory. Set the coloring directory by setting the value of "
+                      "problem.options['coloring_dir'].")
+
+        msg = ["%s: Current coloring configuration does not match the "
+               "configuration of the current model." % obj.msginfo]
+
+        if of_names != self._row_vars:
+            of_diff = set(of_names) - set(self._row_vars)
+            if of_diff:
+                msg.append('The following row vars were added: %s.' % sorted(of_diff))
+            else:
+                of_diff = set(self._row_vars) - set(of_names)
+                if of_diff:
+                    msg.append('The following row vars were removed: %s.' % sorted(of_diff))
+                else:
+                    msg.append('The row vars have changed order.')
+
+        if wrt_names != self._col_vars:
+            wrt_diff = set(wrt_names) - set(self._col_vars)
+            if wrt_diff:
+                msg.append('The following column vars were added: %s.' % sorted(wrt_diff))
+            else:
+                wrt_diff = set(self._col_vars) - set(wrt_names)
+                if wrt_diff:
+                    msg.append('The following column vars were removed: %s.' % sorted(wrt_diff))
+                else:
+                    msg.append('The column vars have changed order.')
+
+        # check sizes
+        changed_sizes = []
+        if of_names == self._row_vars:
+            for i, (my_sz, sz) in enumerate(zip(self._row_var_sizes, of_sizes)):
+                if my_sz != sz:
+                    changed_sizes.append(of_names[i])
+
+        if wrt_names == self._col_vars:
+            for i, (my_sz, sz) in enumerate(zip(self._col_var_sizes, wrt_sizes)):
+                if my_sz != sz:
+                    changed_sizes.append(wrt_names[i])
+
+        if changed_sizes:
+            msg.append('The following variables have changed sizes: %s.' % sorted(changed_sizes))
+
+        if len(msg) > 1:
+            msg.append(msg_suffix)
+            raise RuntimeError(' '.join(msg))
```

Comment:
Would `'n'.join` be better?
Seeing the long unbroken messages in the tests, wonder if line breaks would make it more readable?

Reply:
fixed